app/frontend/src/components/designer/FormDesigner.vue

```diff
@@ -413,6 +607,12 @@ export default {
         this.saving = false;
       }
     },
+    async onUndoClick() {
+      this.undoPatchFromHistory();
+    },
+    async onRedoClick() {
+      this.redoPatchFromHistory();
+    },
```

Comment:
Not all methods need to be declared as async functions. For these ones, you can make these just normal functions. In general, you only want to make these methods async if and only if:

1. The method body has at least one `await`
2. The return result is expected to be of Promise type.
3. You know the methods invoked are asynchronous in nature and cannot behave synchronously.

Double check the rest of your new methods as well.